### PR TITLE
DSRE-695 Remove search-terms restrictions from quicksuggest ETL

### PR DESCRIPTION
**Pull request by jklukas**
See https://mozilla-hub.atlassian.net/browse/DSRE-695

Note that this should not be merged until both https://github.com/mozilla-services/mozilla-pipeline-schemas/pull/732 and https://github.com/mozilla-services/cloudops-infra/pull/4013 are merged and applied.

Also, before this is merged we'll need to make some manual changes to match the state of this PR:

- Run a query to populate the quicksuggest-impression stable table from the sanitized version so that we have 14 days of history to start from
- Rerun init statements for the materialized views changed in this PR
- Drop the tables that are removed in this PR

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

File: CODEOWNERS
Comment (jklukas) [inline CODEOWNERS:4]: We no longer need codeowners on `contextual_services*` paths because there are no longer any search terms under this path.

File: bigquery_etl/shredder/config.py
Comment (jklukas) [inline bigquery_etl/shredder/config.py:294]: This table is removed later in this PR.

File: sql/moz-fx-data-shared-prod/contextual_services/quicksuggest_impression/metadata.yaml
Comment (jklukas) [inline sql/moz-fx-data-shared-prod/contextual_services/quicksuggest_impression/metadata.yaml:19]: This view no longer needs to be authorized because it's been switched to point at a table that will already have access granted to `contextual-services`

File: sql/moz-fx-data-shared-prod/monitoring/suggest_impression_rate_live/view.sql
Comment (jklukas) [inline sql/moz-fx-data-shared-prod/monitoring/suggest_impression_rate_live/view.sql:19]: This complexity is no longer needed because the quicksuggest_impression tables can now have 30 day retention.

**Review by whd — APPROVED** [review]
CI failure is probably https://bugzilla.mozilla.org/show_bug.cgi?id=1775029#c2

Do  I need to do anything other than approve this e.g. delete tables?

**Comment by jklukas** [comment]
> CI failure is probably https://bugzilla.mozilla.org/show_bug.cgi?id=1775029#c2
> 
> Do I need to do anything other than approve this e.g. delete tables?

Yes:

- Delete `moz-fx-data-shared-prod.monitoring_derived.suggest_impression_rate_v1`
- Drop `moz-fx-data-shared-prod.monitoring_derived.suggest_impression_rate_live_v1`
- Reinitialize `moz-fx-data-shared-prod.monitoring_derived.suggest_impression_rate_live_v1` via [init.sql](https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/monitoring_derived/suggest_impression_rate_live_v1/init.sql) 

- Delete `contextual_services_derived.suggest_impression_sanitized_v1`
- Delete `contextual_services.suggest_impression_sanitized`

I'll need to read through this one more time and confirm if any other changes are needed.

**Comment by whd** [comment]
> I'll need to read through this one more time and confirm if any other changes are needed.

Ok once you've confirmed I will land this today. Re-initialization will incur some "downtime" as the authorization will take some time to propagate.

**Comment by jklukas** [comment]
I've updated the instructions in https://github.com/mozilla/bigquery-etl/pull/2908#issuecomment-1162016639 and those are now ready to go.

**Comment by whd** [comment]
Steps have been run but there may be some followup required, see https://mozilla-hub.atlassian.net/browse/DSRE-695?focusedCommentId=560181

**Pull request merged**
